internal/ecsservicediscovery/decoratedtask.go

```diff
@@ -138,6 +140,9 @@ func (t *DecoratedTask) generatePrometheusTarget(
 	addExporterLabels(labels, taskGroupLabel, t.Task.Group)
 	addExporterLabels(labels, taskStartedbyLabel, t.Task.StartedBy)
 	addExporterLabels(labels, taskLaunchTypeLabel, t.Task.LaunchType)
+	if arn, err := arn.Parse(*t.Task.TaskArn); err == nil {
+		addExporterLabels(labels, taskArnResourceLabel, &arn.Resource)
```

Comment:
is there any value to be gained by attempting to break out the parts of the resource such as cluster-name and cluster-instance-id?

Reply:
Personally, I'd really like that, but AFAIK the AWS SDK doesn't provide a way to do it so I settled for this. We could do a string split but that felt a bit dirty, and I was also concerned about ARN format changes. Definitely open to ideas on this one, though

Reply:
I think it could be possible to take the ARN string and then use a regex to capture those values

Reply:
ECS task ARNs are deterministic and AFAIK there is just 2 formats for the resource.  String split on the / should do the trick.  I'm just wondering if you can put a / in the name. I'm going to guess it's not allowed.

Reply:
Quickly perused on AWS console and don't think they allow you to put `/` in any of the names. The only issue is if a user created an ECS resource prior to the ARN change, I don't think their ARN would be updated (opt-out). Based on [docs](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids), there would only be one `/` in the old ARN format so if splitting the string gives us 2 tokens, then we don't have the cluster name in the resource.

Reply:
I think rather than exporting the ARN, we should export the task ID and cluster name (if available) since that's the root issue you're facing. If the cluster name isn't in the ARN, then there's no difference between exporting just the task ID versus exporting the whole ARN. Personally, I like having the `arn.Parse()` call here just as a sanity check to make sure that the ARN passed in is a valid ARN, and would really just want to add additional work inside of the if-block here to extract the task ID and cluster name

Reply:
Sgtm, and I think @SaxyPandaBear particularly nailed it with:

> we should export the task ID and cluster name (if available) since that's the root issue you're facing

Leave it with me and I'll get back to this next week or the week after.

Reply:
Based on the docs here there is an old and new format.  Looks like 2 or 3 parts of a string split by `/`.  Both old and new formats should be supported.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids

Reply:
Done, and both formats are supported. Ready for another review.

Reply:
I'll look through the changes tonight